File: JavaScript/325-Maximum Size Subarray Sum Equals k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxSubArrayLen = function(nums, k) {
    let hmap = new Map();
    hmap.set(0,-1);
    let res = 0, sum = 0;
    for (let i=0; i<nums.length; i++){
        sum += nums[i];
        if (!hmap.has(sum)){
            hmap.set(sum, i);
        }
        if (hmap.has(sum-k)){
            res = Math.max(res, i-hmap.get(sum-k));
        }
    }
    return res;
};

File: C++/325-Maximum Size Subarray Sum Equals k.cpp
using LL = long long;
class Solution {
public:
    int maxSubArrayLen(vector<int>& nums, int k) {
        unordered_map<LL, int> hmap;
        hmap[0] = -1;
        LL curr = 0;
        int res = 0;
        for (int i=0; i<nums.size(); i++) {
            curr += nums[i];
            if (hmap.contains(curr-k)) {
                res = max(res, i-hmap[curr-k]);
            }

            if (!hmap.contains(curr)) {
                hmap[curr] = i;
            }
        }

        return res;
    }
};